tests/integration/test_utils/data_source_config/base.py

```diff
@@ -74,3 +84,70 @@ def _random_resource_name() -> str:
     @cached_property
     def _context(self) -> AbstractDataContext:
         return gx.get_context(mode="ephemeral")
+
+
+class SQLBatchTestSetup(BatchTestSetup, ABC, Generic[_ConfigT]):
+    @property
+    @abstractmethod
+    def connection_string(self) -> str:
+        """Connection string used to connect to SQL backend."""
+
+    @property
+    @abstractmethod
+    def schema(self) -> Union[str, None]:
+        """Schema -- if any -- to use when connecting to SQL backend."""
+
+    @property
+    @abstractmethod
+    def inferrable_types_lookup(self) -> Dict[Type, TypeEngine]:
+        """Dict of Python type keys mapped to SQL dialect-specific SqlAlchemy types."""
+
+    def __init__(self, config: _ConfigT, data: pd.DataFrame) -> None:
+        self.table_name = f"{config.label}_expectation_test_table_{self._random_resource_name()}"
+        self.engine = create_engine(url=self.connection_string)
+        self.metadata = MetaData()
+        self.table: Union[Table, None] = None
+        super().__init__(config, data)
+
+    def get_column_types(self) -> Dict[str, TypeEngine]:
+        column_types = self.infer_column_types()
+        # prefer explicit types if they're provided
+        column_types.update(self.config.column_types or {})
+        untyped_columns = set(self.data.columns) - set(column_types.keys())
+        if untyped_columns:
+            config_class_name = self.config.__class__.__name__
+            message = (
+                f"Unable to infer types for the following column(s): "
+                f"{', '.join(untyped_columns)}. \n"
+                f"Please provide the missing types as the `column_types` "
+                f"parameter when \ninstantiating {config_class_name}."
+            )
+            raise RuntimeError(message)
+        return column_types
+
+    def infer_column_types(self) -> Dict[str, TypeEngine]:
+        inferred_column_types: Dict[str, TypeEngine] = {}
+        for column, value_list in self.data.to_dict("list").items():
+            python_type = type(value_list[0])
```

Comment:
I'd rather merge this and then circle back on this suggestion, but checking just the first item seems a bit off. Pandas looks like it has an api to infer types. No clue on if it works for us. https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.api.types.infer_dtype.html